src/app/catalog/products/products.component.ts

```diff
@@ -14,23 +14,34 @@ export class ProductsComponent implements OnInit, OnDestroy {
   pageSize = 8;
 
   products: Product[] = [];
-  subscription$ = new Subscription();
+  private destroy$ = new Subject<boolean>();
 
   constructor(private filter: FilterService, private http: HttpService) {}
 
   ngOnInit(): void {
-    this.subscription$.add(this.getProduct());
+    this.http
+      .getData()
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((res) => (this.products = res));
+    this.getProduct();
```

Comment:
Why you request product, after request product? Pls, make correct name and move request tot separate method